controllers/ocsinitialization/ocsoperator_config.go

```diff
@@ -0,0 +1,147 @@
+package ocsinitialization
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// This configmap is purely for the OCS operator to use
+	ocsOperatorConfigName = "ocs-operator-config"
+)
+
+// ensureOCSOperatorConfig ensures that the OCS operator config is present in the
+// cluster & has the correct values
+func (r *OCSInitializationReconciler) ensureOCSOperatorConfig(initialData *ocsv1.OCSInitialization) error {
+	ocsOperatorConfig := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ocsOperatorConfigName, Namespace: initialData.Namespace}, ocsOperatorConfig)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// If it does not exist, create the ConfigMap
+			return r.Client.Create(context.TODO(), newOCSOperatorConfig(initialData.Namespace))
+		}
+		return err
+	}
+	resourceVersion := ocsOperatorConfig.ResourceVersion
+
+	// Keep the clustername in sync with the cluster ID
+	err = r.updateClusterName(ocsOperatorConfig)
+	if err != nil {
+		return err
+	}
+	// restart the rook-ceph-operator pod if any change has occured to the configmap
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: ocsOperatorConfigName, Namespace: initialData.Namespace}, ocsOperatorConfig)
+	if err != nil {
+		return err
+	}
+	if resourceVersion != ocsOperatorConfig.ResourceVersion {
```

Comment:
I don't think you need to do all this. You already know in `updateClusterName()` if you had to make any changes, so just use that to determine if you need to restart the operator Pod. Also, since you're only calling it once, it probably doesn't need to be a function. Just restart the operator Pod any time you have to issue an `Update()`.

Reply:
I made this because in the future we might need to add some other values to the configmap just like this cluster-ID, 
And there might be a case where someone may have manually edited the CM via oc edit.

In both cases, the rook-ceph-operator pod should restart to get the new configurations.

Reply:
1. Don't write functions "for the future". Write simple, readable code that will be easy to move into a function *if* it's ever needed.
2. We are *explicitly* not allowing anyone else to edit this ConfigMap. Just add a `Watch()` to the controller for ConfigMaps, preferably filtered by just ours, and we'll get it every time.

Reply:
Addressed this one